### PR TITLE
add available check for northd enpoint

### LINKED ISSUE #2703: Accelerate northd leader transfer when the node is shut down

### PR DESCRIPTION
**Pull request by hongzhen-ma**
- [ ] Make sure you have followed [Kube-OVN Code Style](https://github.com/kubeovn/kube-ovn/blob/master/CODE_STYLE.md).

### What type of this PR
- Bug fixes

### Which issue(s) this PR fixes:
Fixes #2703 

### <samp>🤖 Generated by Copilot at 618b17d</samp>

Improve ovn-northd leader check by using TCP probe. Modify `pkg/ovn_leader_checker/ovn.go` to implement the probe and configure its parameters.

### <samp>🤖 Generated by Copilot at 618b17d</samp>

> _Oh we are the crew of the ovn-northd ship_
> _And we need to check if our leader is fit_
> _We don't just ping the IP, we probe the TCP_
> _With `probe_interval` and `probe_timeout` set_

### <samp>🤖 Generated by Copilot at 618b17d</samp>

*  Add or modify constants and a global variable to configure the TCP connection check for the ovn-northd leader ([link](https://github.com/kubeovn/kube-ovn/pull/2799/files?diff=unified&w=0#diff-a196c025047d1f0ecba3e30efeebb0b7fa9cf82b4df48f5cf452d55a0e5a84abL34-R40))
*  Implement a new function `checkNorthdEpAvailable` to test the TCP connection to the ovn-northd leader using the endpoint IP and port, and return false if the connection fails more than `MaxFailCount` times ([link](https://github.com/kubeovn/kube-ovn/pull/2799/files?diff=unified&w=0#diff-a196c025047d1f0ecba3e30efeebb0b7fa9cf82b4df48f5cf452d55a0e5a84abL271-R292))
*  Modify the function `checkNorthdEpAlive` to call `checkNorthdEpAvailable` after getting the endpoint IP from the Kubernetes API, and return the result of the TCP connection check instead of just the IP existence ([link](https://github.com/kubeovn/kube-ovn/pull/2799/files?diff=unified&w=0#diff-a196c025047d1f0ecba3e30efeebb0b7fa9cf82b4df48f5cf452d55a0e5a84abL287-R310))
*  Modify the function `doOvnLeaderCheck` to call `checkNorthdEpAlive` instead of `checkNorthdSvcValidIP` when checking the ovn-northd leader status, making the leader check more robust and reliable ([link](https://github.com/kubeovn/kube-ovn/pull/2799/files?diff=unified&w=0#diff-a196c025047d1f0ecba3e30efeebb0b7fa9cf82b4df48f5cf452d55a0e5a84abL354-R375))

### CODE REVIEW

File: pkg/ovn_leader_checker/ovn.go [hunk 1]
```diff
@@ -268,7 +272,24 @@ func checkNorthdSvcExist(cfg *Configuration, namespace, svcName string) bool {
...
+	if err != nil {
+		klog.Errorf("failed to connect to northd leader %s, err: %v", ip, err)
+		failCount++
+		if failCount >= MaxFailCount {
+			return false
+		}
+	} else {
+		failCount = 0
+		klog.V(5).Infof("succeed to connect to northd leader %s", ip)
+		_ = conn.Close()
```
Comment (oilbeater) [inline pkg/ovn_leader_checker/ovn.go:287]: Also need close conn when failed

**Review by oilbeater — CHANGES_REQUESTED** [review]

File: pkg/ovn_leader_checker/ovn.go (hunk 1 above)
Comment (hongzhen-ma) [reply pkg/ovn_leader_checker/ovn.go:292]: The conn should be nil when err occurred.

Comment (hongzhen-ma) [reply pkg/ovn_leader_checker/ovn.go:292]: 企业微信截图_6e0a3cf0-a657-43c2-8bf0-4ad64ffa9a74
企业微信截图_ffe0d92b-a79c-42d1-9bcd-b9676647567a

File: pkg/ovn_leader_checker/ovn.go [hunk 2]
```diff
@@ -204,15 +208,20 @@ func checkNorthdActive() bool {
 	return strings.Contains(result, "active")
 }
 
-func stealLock() {
+func stealLock(cfg *Configuration, namespace string) {
+	epIP := getNorthdEpIP(cfg, namespace, "ovn-northd")
+	if epIP == "" {
```
Comment (oilbeater) [inline pkg/ovn_leader_checker/ovn.go:213]: Should use ovn-sb address to steal lock

**Review by oilbeater — CHANGES_REQUESTED** [review]

File: pkg/ovn_leader_checker/ovn.go (hunk 2 above)
Comment (hongzhen-ma) [reply pkg/ovn_leader_checker/ovn.go]: It's really an error. Thanks. Fixed.

**Review by oilbeater — APPROVED** [review]

**Comment by oilbeater** [comment]
Need backport to 1.11

**Pull request merged**